apx/common/src/apx_fileManager.c

```diff
@@ -457,12 +457,20 @@ void apx_fileManager_triggerFileWriteCmdEvent(apx_fileManager_t *self, apx_file_
       }
       else
       {
+         uint8_t fileWriteCmdEventResult;
          memcpy(dataCopy, data, length);
          msg.msgData4 = dataCopy;
          SPINLOCK_ENTER(self->lock);
-         rbfs_insert(&self->ringbuffer,(const uint8_t*) &msg);
+         fileWriteCmdEventResult = rbfs_insert(&self->ringbuffer,(const uint8_t*) &msg);
          SPINLOCK_LEAVE(self->lock);
-         SEMAPHORE_POST(self->semaphore);
+         if (fileWriteCmdEventResult != E_BUF_OK)
+         {
+            APX_LOG_WARNING("[APX_REMOTE_FILE] ringbuffer error. FileWriteCmdEvent lost");
+         }

```

Comment:
You are about to throw away the message since it did not fit into the ring buffer.
Don't forget to release the memory here allocated by apx_allocator_alloc call at line 453.